eng/common/pipelines/templates/steps/prepare-pipelines.yml

```diff
@@ -17,171 +17,176 @@ parameters:
-        --convention up
-        --agentpool Hosted
-        --branch refs/heads/$(DefaultBranch)
-        --patvar PATVAR
-        --set-managed-variables
-        --debug
-        ${{parameters.UPConventionOptions}}
-      displayName: Create UP Pipelines for Public Repository
-      env:
-        PATVAR: $(azuresdk-azure-sdk-devops-pipeline-generation-pat)
-    - script: >
-        $(Pipeline.Workspace)/pipeline-generator/pipeline-generator
-        --organization https://dev.azure.com/azure-sdk
-        --project internal
-        --prefix ${{parameters.Prefix}}
-        --devopspath "\${{parameters.Prefix}}"
-        --path $(System.DefaultWorkingDirectory)/sdk
-        --endpoint Azure
-        --repository ${{parameters.Repository}}
-        --convention tests
-        --agentpool Hosted
-        --branch refs/heads/$(DefaultBranch)
-        --patvar PATVAR
-        --set-managed-variables
-        --debug
-        ${{parameters.TestsConventionOptions}}
-      displayName: Create Live Test Pipelines for Public Repository
-      condition: and(succeeded(), ne('${{parameters.TestsConventionOptions}}',''))
-      env:
-        PATVAR: $(azuresdk-azure-sdk-devops-pipeline-generation-pat)
-    - script: >
-        $(Pipeline.Workspace)/pipeline-generator/pipeline-generator
-        --organization https://dev.azure.com/azure-sdk
-        --project internal
-        --prefix ${{parameters.Prefix}}
-        --devopspath "\${{parameters.Prefix}}"
-        --path $(System.DefaultWorkingDirectory)/sdk
-        --endpoint Azure
-        --repository ${{parameters.Repository}}
-        --convention testsweekly
-        --agentpool Hosted
-        --branch refs/heads/$(DefaultBranch)
-        --patvar PATVAR
-        --set-managed-variables
-        --debug
-        ${{parameters.TestsConventionOptions}}
-      displayName: Create Weekly (Multi-Cloud) Live Test Pipelines for Public Repository
-      condition: and(succeeded(), ne('${{parameters.TestsConventionOptions}}',''))
-      env:
-        PATVAR: $(azuresdk-azure-sdk-devops-pipeline-generation-pat)
-    - script: >
-        $(Pipeline.Workspace)/pipeline-generator/pipeline-generator
-        --organization https://dev.azure.com/azure-sdk
-        --project internal
-        --prefix ${{parameters.Prefix}}
-        --devopspath "\${{parameters.Prefix}}"
-        --path $(System.DefaultWorkingDirectory)/sdk
-        --endpoint Azure
-        --repository ${{parameters.Repository}}
-        --convention upweekly
-        --agentpool Hosted
-        --branch refs/heads/$(DefaultBranch)
-        --patvar PATVAR
-        --set-managed-variables
-        --debug
-        ${{parameters.UPConventionOptions}}
-      displayName: Create Weekly (Multi-Cloud) Unified Test Pipelines for Public Repository
-      condition: and(succeeded(), eq(${{parameters.GenerateUnifiedWeekly}},true))
-      env:
-        PATVAR: $(azuresdk-azure-sdk-devops-pipeline-generation-pat)
+jobs:
```

Comment:
As I hinted at by adding the jobs in here this is now not exactly a steps template. While nothing exactly breaks here this should be moved into the jobs folder but will be a little difficult to transition without breaking all the language repos. 

One way to do this would be to actually create a new template under templates/jobs/prepare-pipelines.yml that adds this new pool info and then includes this steps template. Then we can sync that change to all the repos and then once merged we can update each language specific prepare-pipeline.yml to reference the jobs template instead of the steps template.

Reply:
@weshaggard sorry, I didn't pay enough attention to the fact the directory is named `steps` and there is another one called `jobs` (-;

I introduced PR #4930 with the change you proposed. I will abandon this PR.

Note that I opted for copy-pasting-and-adapting contents of `prepare-pipelines.yml`, instead of making the new "job" version of the file reference the existing "steps" version, to avoid adding unnecessary indirection. I saw our other jobs [in `eng/pipelines/templates/jobs`](https://github.com/Azure/azure-sdk-tools/tree/main/eng/pipelines/templates/jobs) and [in `eng/common/pipelines/templates/job`](https://github.com/Azure/azure-sdk-tools/tree/main/eng/common/pipelines/templates/jobs) also define steps inline, and not by referencing another `.yml` file in `steps` dir.